docs/CONTRIBUTING.md

```diff
@@ -19,33 +19,142 @@ Please submit pull requests in response to open issues. If you have a bug or fea
+1. Docker for Desktop comes with docker-compose installed.
+
+#### MacOS (Sierra 10.12 or above)
+
+1. Get [Docker for Desktop For Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac)
+1. Docker for Desktop comes with docker-compose installed.
+
+#### Windows 10 Home Edition and Windows Subsystem for Linux (WSL)
+
+Docker Desktop for Windows is not available on Home Edition, and you cannot run docker in WSL (Windows Subsystem for Linux). You can get the environment set up using **one** of these three methods:
+
+- Get [Docker Toolbox](https://docs.docker.com/toolbox/toolbox_install_windows/) instead of Docker Desktop for Windows. Make sure that your system can do virtualization and enable virtualization.
+- Set up a virtual machine to run Linux Ubuntu
+  1. Make sure your system can do virtualization and enable it.
+  1. [Download VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+  1. [Follow this helpful youtube tutorial to create a virtual machine with Ubuntu](https://www.youtube.com/watch?v=ThsxqznrgCw&t=401s)
+  1. Use the Linux installation instructions [below](#linux-Ubuntu).
+- Update your home edition to Windows 10 Education Edition
+  1. Download Windows 10 Education Edition from the [Seneca Software Center](https://senecacollege.onthehub.com/WebStore/OfferingDetails.aspx?o=c0bd2c36-a530-e511-940e-b8ca3a5db7a1)
+  1. Update your OS using the installation instructions.
+  1. Use [Windows 10 Education Edition](#Windows-10-Pro,-Enterprise,-or-Education) set up instructions.
+
+#### Linux-Ubuntu
+
+This guide is sourced from the official [Docker-CE](https://docs.docker.com/install/linux/docker-ce/ubuntu/) and [Docker-Compose](https://docs.docker.com/compose/install/) Installation Documentation.
+
+**Install Docker Engine - Community Edition**
+
+1. Update the apt package index: `sudo apt-get update`
+2. Install packages to allow apt to use a repository over HTTPS:
+
+```
+sudo apt-get install \
+    apt-transport-https \
+    ca-certificates \
+    curl \
+    gnupg-agent \
+    software-properties-common
+```
+
+3. Add Docker’s official GPG key: `curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -`
+4. Verify that you now have the key with the fingerprint 9DC8 5822 9FC7 DD38 854A E2D8 8D81 803C 0EBF CD88, by searching for the last 8 characters of the fingerprint: `sudo apt-key fingerprint 0EBFCD88`
+5. Use the following command to set up the stable repository:
+
+```
+sudo add-apt-repository \
+   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
+   $(lsb_release -cs) \
+   stable"
+```
+
+6. Update the apt package index again: `sudo apt-get update`
+7. Install the latest version of Docker Engine community: `sudo apt-get install docker-ce docker-ce-cli containerd.io`
+8. Verify your installation by running `sudo docker run hello-world`. This should print a hello world paragraph that includes a confirmation that Docker is working on your system.
```

Comment:
I don't think we need sudo here.

Reply:
Again, if you are a superuser sure, but if you have just got linux up and running, you won't necessarily be a superuser, and anyone who is will know that they do not need to use sudo.

Reply:
What I'll do is add a note that basically says that if you are not a superuser, then you must use sudo before each of the following commands.

Reply:
I don't think you need to be superuser to run docker. I run docker with my user all the time and it's ok.

Reply:
did u add yourself to the docker group?

Reply:
Yes, and since you added the step to add our user to the docker group, we can run docker without using sudo, right?

Reply:
yes but this step is before you do that. you can't do that until docker is installed. I can remove the check for docker to be installed, but I thought it would help a first time user.

Reply:
I think it's a good idea to have the check for docker to be installed, but maybe you could have the step to add your user to the docker group before the check so we always run docker or docker-compose the same way (without using sudo), otherwise we're adding 2 ways to run docker without really saying why.

Reply:
ok